monkey/agent_plugins/exploiters/rdp/src/rdp_client.py

```diff
@@ -0,0 +1,276 @@
+        target_settings = RDPIOSettings()
+        target_settings.video_width = 1024
+        target_settings.video_height = 768
+        target_settings.video_bpp_max = 32
+        target_settings.video_out_format = VIDEO_FORMAT.PIL
+        target_settings.clipboard_use_pyperclip = False
+        target_settings.clipboard = clipboard
+
+        return target_settings
+
+    def connected(self) -> bool:
+        return self._authenticated
+
+    def execute_command(self, command: str):
+        try:
+            execute_task = self._event_loop.create_task(self._execute_command(command))
+            self._event_loop.run_until_complete(execute_task)
+        except Exception as err:
+            logger.exception(f"Something happened while executing: {err}")
+            self._event_loop.close()
+            raise err
+
+    async def _execute_command(self, command: str):
+        logger.debug("Executing commands...")
+
+        await self._open_windows_run_dialog()
+        # Note: Do we want to grab the output of the command?
+        await self._connection.set_current_clipboard_text(command)
+        await asyncio.sleep(0.5)  # Yield time for the clipboard to sync
+        await self._remote_paste()
+        await self._send_vk_keypress(VK_RETURN)
+
+    async def _open_windows_run_dialog(self):
+        # Open run dialog
+        scancode, _ = self._kb_layout.char_to_scancode("r")
+        modifier_scancode = self._kb_layout.vk_to_scancode("VK_LWIN")
+        await self._send_keypress([modifier_scancode, scancode])
+        await asyncio.sleep(0.5)
+
+    async def _send_keypress(self, scancodes: List[int]):
+        if len(scancodes) == 0:
+            return
+
+        scancode = scancodes[0]
+        await self._connection.send_key_scancode(
+            scancode,
+            is_pressed=True,
+            is_extended=False,
+        )
+        await self._send_keypress(scancodes[1:])
+        await self._connection.send_key_scancode(
+            scancode,
+            is_pressed=False,
+            is_extended=False,
+        )
+
+    async def _send_vk_keypress(self, virtual_key: str):
+        scancode = self._kb_layout.vk_to_scancode(virtual_key)
+        await self._send_keypress([scancode])
+        await asyncio.sleep(0.1)
+
+    async def _send_keys(self, string: str):
+        # Type the string
+        for char in string:
+            if char in self._kb_layout.char_to_sc:
+                scancode, modifiers = self._kb_layout.char_to_scancode(char)
+                if modifiers == VK_MODIFIERS.VK_SHIFT or char.isupper():
+                    modifier_scancode = self._kb_layout.vk_to_scancode("VK_LSHIFT")
+                    await self._send_keypress([modifier_scancode, scancode])
+                else:
+                    await self._send_keypress([scancode])
+            else:
+                logger.warning(f"Character '{char}' not found in layout {self._kb_layout.name}")
+        await asyncio.sleep(0.2)
```

Comment:
Consider inverting some logic to reduce nesting. For example:
```suggestion
    async def _send_keys(self, string: str):
        # Type the string
        for char in string:
            if char not in self._kb_layout.char_to_sc:
                logger.warning(f"Character '{char}' not found in layout {self._kb_layout.name}")
                continue
                
            scancode, modifiers = self._kb_layout.char_to_scancode(char)
            if modifiers == VK_MODIFIERS.VK_SHIFT or char.isupper():
                modifier_scancode = self._kb_layout.vk_to_scancode("VK_LSHIFT")
                await self._send_keypress([modifier_scancode, scancode])
            else:
                await self._send_keypress([scancode])
                
        await asyncio.sleep(0.2)
```